.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,314 @@
+name: 'Setup OpenTofu'
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  tofu-versions:
+    name: 'OpenTofu Versions'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        # TODO: add move versions to test when the feature is added
+        tofu-versions: [latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu - ${{ matrix['tofu-versions'] }}
+      uses: ./
+      with:
+        tofu_version: ${{ matrix['tofu-versions'] }}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] != 'latest' }}
+      run: tofu version | grep ${{ matrix['tofu-versions']}}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] == 'latest' }}
+      run: tofu version | grep 'OpenTofu v'
```

Comment:
dup?

Reply:
Yes, i meant for duplicate, since we also check without the `v`

Reply:
The difference between the command `tofu version | grep ${{ matrix['tofu-versions']}}` and `tofu version | grep 'OpenTofu v'` is that the first one checks for specific version (it's beyond the scope of current PR) when the second is generic. I adjusted the CI: it checks whether OpenTofu was installed, or not probes the version unless it's "latest".